packages/nextra/src/server/mdx-plugins/rehype-better-react-mathjax.ts

```diff
@@ -0,0 +1,134 @@
+import type { ImportDeclaration } from 'estree'
+import { valueToEstree } from 'estree-util-value-to-estree'
+import type { Element, Root, RootContent } from 'hast'
+import type { MdxJsxAttribute } from 'hast-util-to-estree/lib'
+import type { Plugin } from 'unified'
+import { visit } from 'unist-util-visit'
+import type { MathJaxOptions } from '../../types'
+
+const MATHJAX_IMPORTS = {
+  type: 'mdxjsEsm',
+  data: {
+    estree: {
+      body: [
+        {
+          type: 'ImportDeclaration',
+          source: { type: 'Literal', value: 'nextra/components' },
+          specifiers: ['MathJax', 'MathJaxContext'].map(name => ({
+            type: 'ImportSpecifier',
+            imported: { type: 'Identifier', name },
+            local: { type: 'Identifier', name }
+          }))
+        } satisfies ImportDeclaration
+      ]
+    }
+  }
+}
+
+function wrapInMathJaxContext(
+  children: RootContent[],
+  { config, src }: NonNullable<MathJaxOptions>
+) {
+  const attributes: MdxJsxAttribute[] = []
+  if (src) {
+    attributes.push({ type: 'mdxJsxAttribute', name: 'src', value: src })
+  }
+  if (config && Object.keys(config).length) {
+    attributes.push({
+      type: 'mdxJsxAttribute',
+      name: 'config',
+      value: {
+        type: 'mdxJsxAttributeValueExpression',
+        value: '',
+        data: {
+          estree: {
+            type: 'Program',
+            sourceType: 'module',
+            body: [
+              { type: 'ExpressionStatement', expression: valueToEstree(config) }
+            ]
+          }
+        }
+      }
+    })
+  }
+
+  return {
+    type: 'mdxJsxFlowElement',
+    name: 'MathJaxContext',
+    attributes,
+    children
+  }
+}
+
+/**
+ * Wrap the math in the appropriate braces. Defaults to `\(...\)` for inline and `\[...\]` for display,
+ * but will use the braces provided by `options` if they are present.
+ */
+function wrapInBraces(
+  source: string,
+  mathInline: boolean,
+  options: NonNullable<MathJaxOptions>
+): string {
+  const { inlineMath, displayMath } = options.config?.tex || {}
+
+  const inlineBraces = inlineMath?.[0] || ['\\(', '\\)']
+  const displayBraces = displayMath?.[0] || ['\\[', '\\]']
+  const [before, after] = mathInline ? inlineBraces : displayBraces
+  return `${before}${source}${after}`
+}
```

Comment:
@siefkenj I am just curious why we need this? 😅

Reply:
I was trying to make things behave more as the user would expect. For example, if they set `inlineMath: [['..', '..']]` in the options and then did

```
<MathJax>
Some ..math..
</MathJax>
```
then `math` would be the only thing rendered as math.